core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -97,7 +107,9 @@ public final void initialize(TelemetryConfiguration configuration) {
 
             initializeComponents(configuration);
         } catch (Exception e) {
-            InternalLogger.INSTANCE.error("Failed to initialize configuration, exception: %s", e.getMessage());
+            InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.ERROR, "Failed to initialize configuration, exception: %s", e.getMessage());
+        } finally {
+            InternalLogger.INSTANCE.configMode(false);
```

Comment:
See Yoni's and my comments about using `configMode`. Also, if you decide to stick with it please use java convention, i.e. `setConfigMode`